src/content/1.7/modules/sample_modules/grid-and-identifiable-object-form-hooks-usage.md

```diff
@@ -0,0 +1,227 @@
+---
+title: Grid and identifiable object form hooks usage example
+weight: 10
+---
+
+# Grid and identifiable object form hooks usage example
+{{< minver v="1.7.6" title="true" >}}
+
+## Introduction
+
+In this tutorial we are going to build module which extends customers list with one extra column
+which can be toggled. It can have two states - turned on or off. In customer creation and edit
+form we will add switch which will also manage the same state. By following this tutorial you will learn
+how to:
+
+- extend modern grids. [Grid component]({{< relref "/1.7/development/components/grid/_index.md" >}}) 
+- extend identifiable object form. [identifiable object form]({{< relref "CRUD-forms.md" >}}) 
+
+The module created within this tutorial can be found [here](https://github.com/friends-of-prestashop/demo-cqrs-hooks-usage-module)
+
+## Prerequisites
+
+- To be familiar with basic module creation.
+
+### Register hooks
+
+On module installation the following hooks are being registered:
+
+- `action`**Customer**`GridDefinitionModifier` - for adding new column to customers grid.
+- `action`**Customer**`GridQueryBuilderModifier` - for modifying customers grid sql.
+- `action`**Customer**`FormBuilderModifier` - for adding new field to customers create or edit form field.
+- `actionAfterCreate`**Customer**`FormHandler` - to execute the saving process of added field from the module.
+- `actionAfterUpdate`**Customer**`FormHandler` - to execute the update process of added field from the module.
+
+```php
+
+public function install()
+{
+    return parent::install() &&
+        $this->registerHook('actionCustomerGridDefinitionModifier') &&
+        $this->registerHook('actionCustomerGridQueryBuilderModifier') &&
+        $this->registerHook('actionCustomerFormBuilderModifier') &&
+        $this->registerHook('actionAfterCreateCustomerFormHandler') &&
+        $this->registerHook('actionAfterUpdateCustomerFormHandler') &&
+        $this->installTables()
+    ;
+}
+```
+
+{{% notice note %}}
+Hooks `action`**Customer**`GridDefinitionModifier` and `action`**Customer**`GridQueryBuilderModifier` are built
+using grid id which in this case is **Customer**. If You want to use hook for any other grid 
+You can check any definition factory service in `PrestaShop\PrestaShop\Core\Grid\Definition\Factory` to see available grid ids. Grid id is returned by `getId()` method.
+<br />
+<br />
+Hooks `action`**Customer**`FormBuilderModifier`,`actionAfterCreate`**Customer**`FormHandler` and `actionAfterUpdate`**Customer**`FormHandler` unique identifier which
+in this case is **Customer** is retrieved from its form type **CustomerType**. E.g id retrieved from **ManufacturerType** will be **Manufacturer**. Some types might use
+function `getBlockPrefix` to retrieve the unique id
+{{% /notice %}}
+
+### Adding new column to customers grid
+
+#### Extending grid definition and filters
+
```

Comment:
Please explain that this hook receives through `$params` a `GridDefinition` that defines how the grid is rendered (refer to grid doc) and can be modified to add/remove/modify any item of the grid